addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/JsoScanner.java

```diff
@@ -0,0 +1,137 @@
+    private static final String JSO_BASE_64_MAGIC_SEQUENCE = "rO0AB";
+    private PassiveScanThread parent;
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        // do nothing
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        checkJsoInHeaders(msg);
+
+        checkJsoInCookies(msg);
+
+        checkJsoInBody(msg);
+    }
+
+    private void checkJsoInBody(HttpMessage msg) {
+        byte[] startOfBody =
+                Arrays.copyOfRange(
+                        msg.getResponseBody().getBytes(), 0, JSO_BYTE_MAGIC_SEQUENCE.length);
+        String startOfBodyAsString =
+                new String(
+                        Arrays.copyOfRange(
+                                msg.getResponseBody().getBytes(),
+                                0,
+                                JSO_BASE_64_MAGIC_SEQUENCE.length()),
+                        StandardCharsets.UTF_8);
+        if (Arrays.equals(JSO_BYTE_MAGIC_SEQUENCE, startOfBody)
+                || hasJsoBase64MagicSequence(startOfBodyAsString)) {
+            Alert alert = buidAlert(msg, "");
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private void checkJsoInCookies(HttpMessage msg) {
+        for (HttpCookie cookie : msg.getResponseHeader().getHttpCookies(null)) {
+            String value = cookie.getValue();
+            if (!hasJsoBase64MagicSequence(value)) {
+                continue;
+            }
+
+            Alert alert = buidAlert(msg, cookie.toString());
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private boolean hasJsoBase64MagicSequence(String value) {
+        return value.startsWith(JSO_BASE_64_MAGIC_SEQUENCE);
+    }
+
+    private void checkJsoInHeaders(HttpMessage msg) {
+        for (HttpHeaderField header : msg.getResponseHeader().getHeaders()) {
+            String value = header.getValue();
+            if (!hasJsoBase64MagicSequence(value)) {
+                continue;
+            }
+
+            Alert alert = buidAlert(msg, header.toString());
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private Alert buidAlert(HttpMessage msg, String evidence) {
+        Alert alert =
+                new Alert(getPluginId(), Alert.RISK_MEDIUM, Alert.CONFIDENCE_MEDIUM, getName());
+        alert.setDetail(
+                Constant.messages.getString(MESSAGE_PREFIX + "desc"),
+                msg.getRequestHeader().getURI() + "",
+                null,
+                "",
+                "",
+                getString("soln"),
+                getString("refs"),
+                evidence,
+                502, // CWE-502: Deserialization of Untrusted Data
+                -1, // No WASC-ID
+                msg);
+        return alert;
+    }
+
+    private String getString(String param) {
+        return Constant.messages.getString(MESSAGE_PREFIX + param, "");
```

Comment:
Call `I18N.getString(String)` (there are no parameters in the resource messages).